NextcloudTalk/UserProfileTableViewControllerAvatarSetup.swift

```diff
@@ -0,0 +1,129 @@
+        // https://stackoverflow.com/a/20464727/2512312
+        let mediaType: String = AVMediaType.video.rawValue
+        let authStatus = AVCaptureDevice.authorizationStatus(for: AVMediaType(mediaType))
+        if authStatus == AVAuthorizationStatus.authorized {
+            self.presentCamera()
+            return
+        } else if authStatus == AVAuthorizationStatus.notDetermined {
+            AVCaptureDevice.requestAccess(for: AVMediaType(mediaType)) { granted in
+                if granted {
+                    self.presentCamera()
+                }
+            }
+            return
+        }
+        let alertTitle = NSLocalizedString("Could not access camera", comment: "")
+        let alertMessage = NSLocalizedString("Camera access is not allowed. Check your settings.", comment: "")
+        let alert = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+        let okButton = UIAlertAction(title: NSLocalizedString("OK", comment: ""), style: .default, handler: nil)
+        alert.addAction(okButton)
+        NCUserInterfaceController.sharedInstance().presentAlertViewController(alert)
+    }
+
+    func presentCamera() {
+        DispatchQueue.main.async {
+            self.imagePicker = UIImagePickerController()
+            self.imagePicker?.sourceType = UIImagePickerController.SourceType.camera
+            if self.imagePicker != nil {
+                self.imagePicker!.delegate = self
+                self.present(self.imagePicker!, animated: true)
+            }
+        }
+    }
+
+    func presentPhotoLibrary() {
+        DispatchQueue.main.async {
+            self.imagePicker = UIImagePickerController()
+            self.imagePicker?.sourceType = .photoLibrary
+            if self.imagePicker != nil {
+                self.imagePicker!.delegate = self
+                self.present(self.imagePicker!, animated: true)
+            }
+        }
+    }
+
+    func sendUserProfileImage(image: UIImage) {
+        NCAPIController.sharedInstance().setUserProfileImage(image, for: account) { error, _ in
+            if error == nil {
+                self.refreshUserProfile()
+            } else {
+                self.showProfileImageError(true)
+                print("Error removing profile image: \(error.debugDescription)")
+            }
+        }
+    }
+
+    func removeUserProfileImage() {
+        NCAPIController.sharedInstance().removeUserProfileImage(for: account) { error, _ in
+            if error == nil {
+                self.refreshUserProfile()
+            } else {
+                self.showProfileImageError(false)
+                print("Error removing profile image: ", error.debugDescription)
+            }
+        }
+    }
+
+    func showProfileImageError(_ setting: Bool) {
+        let reason = setting ? NSLocalizedString("An error occurred setting profile image", comment: "") : NSLocalizedString("An error occurred removing profile image", comment: "")
```

Comment:
Wouldn't it make it easier to understand, when the reson is the parameter of `showProfileImageError` instead of a boolean?